plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/DecimalCoercers.java

```diff
@@ -228,6 +232,46 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
         }
     }
 
+    public static Function<Block, Block> createDecimalToVarcharCoercer(DecimalType fromType)
+    {
+        if (fromType.isShort()) {
+            return new ShortDecimalToVarcharCoercer(fromType);
+        }
+        return new LongDecimalToVarcharCoercer(fromType);
+    }
+
+    private static class ShortDecimalToVarcharCoercer
+            extends TypeCoercer<DecimalType, VarcharType>
+    {
+
+        protected ShortDecimalToVarcharCoercer(DecimalType fromType)
+        {
+            super(fromType, VARCHAR);
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            toType.writeString(blockBuilder, BigDecimal.valueOf(fromType.getLong(block, position), fromType.getScale()).toPlainString());
+        }
+    }
+
+    private static class LongDecimalToVarcharCoercer
+            extends TypeCoercer<DecimalType, VarcharType>
+    {
+
+        protected LongDecimalToVarcharCoercer(DecimalType fromType)
+        {
+            super(fromType, VARCHAR);
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            toType.writeString(blockBuilder, Decimals.toString((Int128) fromType.getObject(block, position), fromType.getScale()));
```

Comment:
The first commit doesn't check the varchar length, and is run for any varchar, so it's actually not correct. It can yield a bogus value (too long).

The simplest recourse is probably squashing the commits.